app/src/main/java/com/gh4a/activities/FileViewerActivity.java

```diff
@@ -343,4 +306,38 @@ private static String highlightImage(String imageUrl, String cssTheme, String ti
         content.append("</div></body></html>");
         return content.toString();
     }
+
+    private void loadFile(boolean force) {
+        RepositoryContentService service =
+                Gh4Application.get().getGitHubService(RepositoryContentService.class);
+        service.getContents(mRepoOwner, mRepoName, mPath, mRef)
+                .map(ApiHelpers::throwOnFailure)
+                .onErrorResumeNext(error -> {
+                    if (error instanceof ApiRequestException) {
+                        ClientErrorResponse response = ((ApiRequestException) error).getResponse();
+                        List<ClientErrorResponse.FieldError> errors =
+                                response != null ? response.errors() : null;
+                        if (errors != null) {
+                            for (ClientErrorResponse.FieldError fe : errors) {
+                                if (fe.reason() == ClientErrorResponse.FieldError.Reason.TooLarge) {
+                                    openUnsuitableFileAndFinish();
+                                    return Single.just(null);
```

Comment:
`Single.just` doesn't accept null as value. We might want to use `Maybe.empty()` instead, though that will require some more changes to work with `Maybe`.